server/src/infra/repositories/search.repository.ts

```diff
@@ -249,4 +260,20 @@ export class SearchRepository implements ISearchRepository {
 
     return runtimeConfig;
   }
+
+  @GenerateSql({ params: [DummyValue.STRING] })
+  async searchPlaces(placeName: string): Promise<GeodataPlacesEntity[]> {
+    return await this.geodataPlacesRepository
+      .createQueryBuilder('geoplaces')
+      .leftJoinAndSelect('geoplaces.admin1', 'admin1')
+      .leftJoinAndSelect('geoplaces.admin2', 'admin2')
+      .where(
+        '(LOWER(geoplaces.name) LIKE :nameStart OR LOWER(geoplaces.name) LIKE :nameAnywhere OR LOWER(admin1.name) LIKE :nameStart OR LOWER(admin1.name) LIKE :nameAnywhere OR LOWER(admin2.name) LIKE :nameStart OR LOWER(admin2.name) LIKE :nameAnywhere)',
+        { nameStart: `${placeName.toLowerCase()}%`, nameAnywhere: `% ${placeName.toLowerCase()}%` },
+      )
```

Comment:
I'm looking at optimizing this. We should add a [trigram indices](https://about.gitlab.com/blog/2016/03/18/fast-search-using-postgresql-trigram-indexes/) for the names and use `~* '\m<insert_text_here>'` for the comparison.

Reply:
Done

Reply:
Nice, thank you !

The logic was a bit dumb. I want to improve it too by passing multiple input, so like if you type "France, Paris" this string will be split into "Paris" and "France". I want it to be accents and characters insensitive too.

Do you have any idea how to achieve this ?

Reply:
The new query is accent and case insensitive, so we should be good to go there. "France, Paris" is a little trickier since the individual `name`, `admin1Name` and `admin2Name` fields are searched separately. I tweaked the query so it can handle that kind of input. It might need some more tweaking to get the ranking right.

Reply:
Hey, maybe we can extract the population to have a better sort. So when you search Washington, you prioritize the cities, with the most inhabitants

Reply:
Oh we can even prioritize by features code (like capitals, populated places, airports ...)